### ISSUE TITLE
Vite can't parse "storybook-react-i18next" dependence "storybook-i18n"

### ISSUE THREAD

**Issue by Cyberhan123** [eirslett/storybook-builder-vite#233]
<img width="1179" alt="截屏2022-02-10 下午11 32 03" src="https://user-images.githubusercontent.com/38837039/153441353-bb93b23d-1fe1-439c-9c5b-bb2c9ba080ec.png">

And In preview.js 
```js
module.exports = require("./dist/cjs/preset/preview");
```
So It throw an error 
ReferenceError: require is not defined

And This code at
[    const getProjectAnnotations = async () =>](https://github.com/eirslett/storybook-builder-vite/blob/01bfbe6767e7ce8abab3b3916dcd11d6b82cc573/packages/storybook-builder-vite/codegen-modern-iframe-script.ts#L43)

What should I do

**Comment by Cyberhan123** [comment]
> Yes, I think this bug needs to be fixed in these libraries. They're not 100 % ESM-compatible, it appears. One could possibly do some trickery with optimizeDeps, I'm not sure...
Yeah,  I have no idea about optimizeDeps with ESM-compatible，can you give me some example...

**Comment by jrmyio** [comment]
Did anyone find a solution or workaround?
